src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COST;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddTransactionCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Name;
+import seedu.address.model.transaction.Amount;
+import seedu.address.model.transaction.Description;
+import seedu.address.model.transaction.Timestamp;
+import seedu.address.model.transaction.Transaction;
+import seedu.address.model.transaction.expense.Expense;
+import seedu.address.model.transaction.expense.Weight;
+
+/**
+ * Parses input arguments and creates a new AddTransactionCommand object
+ */
+public class AddTransactionCommandParser implements Parser<AddTransactionCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddTransactionCommand
+     * and returns an AddTransactionCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddTransactionCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_COST, PREFIX_DESCRIPTION);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_COST, PREFIX_DESCRIPTION)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddTransactionCommand.MESSAGE_USAGE));
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_COST, PREFIX_DESCRIPTION);
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        Amount amount = ParserUtil.parseAmount(argMultimap.getValue(PREFIX_COST).get());
+        Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get());
+        Timestamp timestamp = Timestamp.now();
+        Weight weight = new Weight("1");
+        Expense expense = new Expense(name, weight);
+        Set<Expense> expensesSet = Set.of(expense);
+        Transaction transaction = new Transaction(amount, description, Name.SELF, expensesSet, timestamp);
```

Comment:
So addTransaction only works if we lend money / paid first, and not if we borrow money?

Reply:
Actually not sure, guess can discuss tmr. But rn ig this works for 1.2?